interface-definitions/interfaces_openvpn.xml.in

```diff
@@ -318,6 +318,29 @@
             </properties>
             <defaultValue>udp</defaultValue>
           </leafNode>
+          <leafNode name="ip-version">
+            <properties>
+              <help>Force OpenVPN to use a specific IP protocol version</help>
+              <completionHelp>
+                <list>ipv4 ipv6 ipv6-dual-stack</list>
```

Comment:
Do we really need an explicit dual-stack option? Dual stack is the default and I believe it's the default people would expect. If we add an explicit option, we'll have an option that has no effect compared to no option, and I think we should avoid that.

Reply:
There is subtle difference between the default behavior and "dual stack" that is frankly a little bit hard to articulate (obviously some work is needed to make this better self-documenting).

The default behavior should be thought of as "one stack (auto-select)". OpenVPN will, based on other options, DNS resolution, etc., pick a single IP stack to communicate on. What it will not do is accept traffic on both stacks simultaneously, in a single process.

In client mode or site-to-site mode[^1] where there should really only ever be a single connection flow at any given time, this distinction is relatively moot. In server mode however, where multiple clients (both from IPv4 only networks and IPv6 only networks) may connect simultaneously to a single OpenVPN daemon, the distinction is relevant.

[^1]: In site-to-site mode where you are a passive listener, e.g. `tcp-passive`, it could actually be relevant.

Reply:
Also note that "dual-stack" mode is implemented as OpenVPN listening on an IPv6 socket (it does not open IPv4 and IPv6 sockets simultaneously). The reason it can accept both is because the kernel performs a default NAT behavior of v4 addresses into v6 addresses (e.g. `192.168.1.1` becomes `::ffff:192.168.1.1`). This makes it not backwards-compatible if users have client-specific overrides based on the client's IPv4 address, hence it probably should not be the default.

Reply:
It is probably better to use `dual-stack` instead of `ipv6-dual-stack` 
As dual-stack means both, IMHO

Reply:
Thoughts on the following solution?
I will introduce an additional `auto-select` option and make that the default (as opposed to the current implicit default when the value is not set). Then rename `ipv6-dual-stack` to achieve one of the following naming schemes:

```
auto-select: Select the protocol to use based on configured/resolved hosts
ipv4: Bind to or connect to IPv4 addresses only
ipv6: Bind to or connect to IPv6 addresses only
dual-stack: Accept connections on both protocols simultaneously (only supported in server mode)
```

An alternative scheme using single words could also be used: `auto ipv4 ipv6 both`

Reply:
I tend to favor: `auto ipv4 ipv6 both`. 

IIRC (from what I remember in https://vyos.dev/T6183)  `auto` would be OpenVPNs default option, thus please make it a `defaultValue` in the XML code, too.

Thanks for picking up on this task, I will re-assign it to your so you get the credits.